src/jit/morph.cpp

```diff
@@ -12461,7 +12461,7 @@ GenTreePtr Compiler::fgMorphSmpOp(GenTreePtr tree, MorphAddrContext* mac)
                         // We need to keep op1 for the side-effects. Hang it off
                         // a GT_COMMA node
 
-                        tree->ChangeOper(GT_COMMA);
+                        tree->ChangeOper(GT_COMMA, GenTree::PRESERVE_VN);
```

Comment:
As I understand we can save vn if expressions results are equal.
I think that result of (op1, 0) is equal to result of (op1 * 0).
Am I wrong?